server/views/pages/overview/partials/timelineTab/timelineTabContents.njk

```diff
@@ -2,57 +2,26 @@
   <div class="govuk-grid-row">
     <div class="govuk-grid-column-two-thirds app-u-print-full-width">
 
-      {% if timeline.events %}
+      {% if timeline.events | length %}
 
         <div class="moj-timeline">
-
-        {% for event in timeline.events %}
```

Comment:
The logic in this template (should never have been implemented this way) and was difficult to follow and reason about. It would also have become  even messier when we add the markup for the 2 new event types (goal archived and goal unarchived)

We now loop through the array of timeline events, and for each on that is of a supported type we include it's component.